src/Messages/OWSMessageSender.m

```diff
@@ -933,12 +913,10 @@ - (void)handleMismatchedDevices:(NSDictionary *)dictionary
 
 - (void)handleMessageSentLocally:(TSOutgoingMessage *)message
 {
-    [self saveMessage:message withState:TSOutgoingMessageStateSent];
```

Comment:
In https://github.com/WhisperSystems/SignalServiceKit/pull/167 we took a big step towards only marking messages as failed or succeeded in a single place and only when that state transition was "final".  It turns out there was more work to be done.  We now use the OWSSendMessageOperation's success and failure handlers to mark success and failure.